### PR TITLE
feat(core): prevent the floating sidebar from showing immediately after the sidebar is closed

### PR DESCRIPTION
**Pull request by JimmFly**
close PD-1763

https://github.com/user-attachments/assets/a8346098-5f87-4297-bdb6-885f5486ce78

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#8531** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8531?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by pengx17 — COMMENTED** [review]
I think you can add a timer in `AppSidebar` entity to prevent hovering = true when sidebar is just closed

**Comment by JimmFly** [comment]
> I think you can add a timer in `AppSidebar` entity to prevent hovering = true when sidebar is just closed

Maybe it would be better to use the timer as little as possible?

File: packages/frontend/core/src/modules/app-sidebar/views/sidebar-header/sidebar-switch.tsx [hunk 1]
```diff
@@ -16,15 +16,23 @@ export const SidebarSwitch = ({
 }) => {
   const appSidebarService = useService(AppSidebarService).sidebar;
   const open = useLiveData(appSidebarService.open$);
+  const preventHovering = useLiveData(appSidebarService.preventHovering$);
   const switchRef = useRef<HTMLDivElement>(null);
 
   const handleMouseEnter = useCallback(() => {
+    if (preventHovering) {
```
Comment (EYHN) [inline packages/frontend/core/src/modules/app-sidebar/views/sidebar-header/sidebar-switch.tsx]: You can start listening for the mouseenter event after the sidebar is collapsed, which should achieve the same effect without adding additional states.

**Review by EYHN — APPROVED** [review]

**Pull request merged**